File: keevo/app/lib/features/onboarding/domain/model/onboarding_result.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'onboarding_result.freezed.dart';

/// OnboardingResult — Domain model for a successful onboarding response.
///
/// Immutable Freezed value object returned by [CompleteOnboardingUseCase].
/// Mirrors the backend OnboardingResult record.
@freezed
class OnboardingResult with _$OnboardingResult {
  const factory OnboardingResult({
    required String tenantId,
    required String sectorType,
    required String storeName,
    required int categoriesCreated,
  }) = _OnboardingResult;
}

File: keevo/backend/src/main/java/com/keevo/identity/onboarding/domain/port/in/OnboardingResult.java
package com.keevo.identity.onboarding.domain.port.in;

import com.keevo.identity.onboarding.domain.model.SectorType;

/**
 * OnboardingResult — Output of the CompleteOnboardingUseCase.
 *
 * <p>Pure Java record — immutable, contains tenant info and category count.
 */
public record OnboardingResult(
        String tenantId,
        SectorType sectorType,
        String storeName,
        int categoriesCreated
) {}
